### ISSUE TITLE
Default Activity not found

### ISSUE THREAD

**Issue by winster** [googlesamples/android-CommitContentSampleIME#2]
Since there is no activity defined in the manifest, getting error in Android studio.

Error running app:
Default Activity not found

**Comment by tjohns** [comment]
Yes, this is correct. This is an input manager, not an activity.

You can get Android Studio not to complain by going to the "Edit Configurations" menu (tap "Shift" three times, type "Edit Configurations"), then change Launch Options > Launch to "Nothing".

I'll probably add a generic activity containing instructions, just to avoid any confusion.

**Comment by 111usman** [comment]
Thanks a lot

**Comment by letaii** [comment]
thanks you @tjohns

**Comment by neilakoh** [comment]
I also got the same error; however, in my case it was straightforward since this error occurred in a fresh project and it turns out I have to add an intent inside activity tag to get rid of the error:

```
<activity android:name=".MainActivity">
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
```

See reference:

https://www.aboutonline.info/2018/08/error-running-app-default-activity-not-found-on-android-with-kotlin.html

**Comment by AndroidNinza** [comment]
@tjohns i agree with you that if we change that setting then app works fine but what if want to set run app without that `Edit Configuration` ? 
I have spend almost 5 hours and got the solution. It is work for me and i hope some one will get help with this
**Solution :**
Delete the cash from your %Home%\.gradle\cash and sync project again.

**Comment by hishark** [comment]
@AndroidNinza thank you!!!! I spend one night to find solution, I'm trying your solution and it is downloading now, hope I can solve it and then go to bed:'-(

**Comment by hishark** [comment]
@AndroidNinza done!!!!!!!! thank you!!!!!!!!! 😭 😭 😭 
perfect solution

**Comment by moyarich** [comment]
@IDFDeveloper thank you very
>     1. Delete the caches from your `%Home%\.gradle\caches` and sync project again.

that worked really well for me. I tried  file>invalidate caches/restart and that did not work for me

**Comment by awebdesigner** [comment]
I lost over 6 hours hunting down the problem. I even redownload gradle, restarted Android studio, Invalidated cache/restart to no vail. Until a sister pointed me to this wonderful point.

This came to my rescue.

> 
> 
> @tjohns i agree with you that if we change that setting then app works fine but what if want to run the app without edit `Edit Configuration` ? I have checked this [stack overflow question's answers](https://stackoverflow.com/questions/15825081/error-default-activity-not-found) to solve this issue in `android studio 3.2`
> I have spend almost 5 hours and got the solution. I have copy that project and run in another system then its working fine, So just do following solution to fix this issue and it is work for me and i hope some one will get help with this.
> **Solution :**
> 
>     1. Delete the cash from your `%Home%\.gradle\cash` and sync project again.
> 
>     2. Also check that is there in no any duplicate tag in your Manifest.

**Comment by yashkatkar** [comment]
I was Scrathing GOOGLE for this answer 
and at last Found
https://github.com/googlesamples/android-CommitContentSampleIME/issues/2#issuecomment-428936292

**Comment by Johnyoat** [comment]
I had to delete `%Home%\.gradle` , `%Home%\AndroidStudio3.2` before i got it to work.

**Comment by YimasoftLiuGuanJun** [comment]
> I had to delete `%Home%\.gradle` , `%Home%\AndroidStudio3.2` before i got it to work.

me too

**Comment by andreirusu99** [comment]
Make sure your application `label` and your activity `label` are not the same string resource when declared in the `Android Manifest`

**Comment by PauloVilarinho** [comment]
how do i delete this %home%\.gradle\cash im not finding it.

**Comment by AjitPGaikwad** [comment]
> Make sure your application `label` and your activity `label` are not the same string resource when declared in the `Android Manifest`

This is accepted answer in my case

**Comment by zoro238** [comment]
how do i delete this %home%.gradle\cash i am not finding it.

**Comment by ankitk20** [comment]
Go to your users home directory and there you will find _.gradle_ folder

**Comment by hawkraph** [comment]
I deleted android studio cache;
path: C:\Users\user\.AndroidStudio3.3\system\caches
Close android studio, delete cache. start android studio

**Comment by hawkraph** [comment]
I just noticed deleting cache also solve my XML autocompletion of attributes problem

**Comment by shoaibmushtaq25** [comment]
i had these issues with my project:

1. **Default activity not found**
2. xml intellisense was not working
3. kotlin standard function not detecting 
All my above issues were resolved by **Deleting System cache of Android Studio 3.3 at the home path**, and it's working nicely for me,,
**Steps:**

1. exit Android Studio
2. Go to path > C:\Users\<YOUR_WINDOW_USER_NAME>\.AndroidStudio3.3\system
3. Then you have a `\caches` folder, delete this caches folder
4. Now open Android Studio 
Worked for me.. i wasted couple of hours resolving this issue and finally it got resolved in this way.

**Comment by benja420** [comment]
tanks this resolved my problem mexico....

**Comment by Endless-explorer** [comment]
i dont think this is a good sulution，we need the luncher and we config them but it cant run ,so we should find why the problem happen!

**Comment by fatihgokcee** [comment]
> @tjohns i agree with you that if we change that setting then app works fine but what if want to run the app without edit `Edit Configuration` ? I have checked this [stack overflow question's answers](https://stackoverflow.com/questions/15825081/error-default-activity-not-found) to solve this issue in `android studio 3.2`
> I have spend almost 5 hours and got the solution. I have copy that project and run in another system then its working fine, So just do following solution to fix this issue and it is work for me and i hope some one will get help with this.
> **Solution :**
> 
> 1. Delete the cash from your `%Home%\.gradle\cash` and sync project again.
> 2. Also check that is there any duplicate tag in your Manifest.

Thats worked. Thx dude !! Adamın dibisin..

**Comment by SilawatAamil** [comment]
> @tjohns i agree with you that if we change that setting then app works fine but what if want to run the app without edit `Edit Configuration` ? I have checked this [stack overflow question's answers](https://stackoverflow.com/questions/15825081/error-default-activity-not-found) to solve this issue in `android studio 3.2`
> I have spend almost 5 hours and got the solution. I have copy that project and run in another system then its working fine, So just do following solution to fix this issue and it is work for me and i hope some one will get help with this.
> **Solution :**
> 
> 1. Delete the cash from your `%Home%\.gradle\cash` and sync project again.
> 2. Also check that is there any duplicate tag in your Manifest.

Thank you Soo much you save my project 👍

**Comment by inrakesh** [comment]
**@IDFDeveloper thank u so much !!!!! You save my day.**
I tried file>invalidate caches/restart and that did not work for me

**Comment by suniltn** [comment]
I had a similar issue, I was using pixel 2 real device as a test device to install App. in this device I had enabled multiple users (Guest) . I removed the guest user and everything started to work as usual.
